docs/reference/configuration.adoc

```diff
@@ -165,47 +165,81 @@ $ java -jar myapp-swarm.jar -Scloud -Stesting -s/home/app/openshift.yml
+Unlike other configuration options, properties defined as environment variables in Linux-based containers do not allow defining non-alphanumeric characters like _dot_ (.), _dash/hyphen_ (-) or any other characters not in the `[A-Za-z0-9_]` range.
+Many configuration properties in WildFly Swarm contain these characters, so you must follow these rules when defining the environment variables in the following environments:
 
-Linux based container Rules
-====
-* It is a naming convention that all environment properties are defined using UPPERCASE letters. For example: property `serveraddress` is defined as `SERVERADDRESS`. 
-* All the dot(.) characters MUST to be replaced with underscore (_). For example: property `swarm.bind.address=127.0.0.1` should be defined as `SWARM_BIND_ADDRESS=127.0.0.1`
-* All dash(-) characters MUST be replaced with string "__DASH__". For example: a property `swarm.data-sources.foo.url=<url>` needs to be `SWARM_DATA_DASH_SOURCES_FOO_URL=<url>`
-* If the property name has underscores then all underscore(-) characters MUST be replaced with string "__UNDERSCORE__". For example: a property `swarm.data_sources.foo.url=<url>` needs to be `SWARM_DATA_UNDERSCORE_SOURCES_FOO_URL=<url>`
-=====
+.Linux-based container rules
+* It is a naming convention that all environment properties are defined using uppercase letters.
+For example, define the `serveraddress` property as `SERVERADDRESS`.
+* All the _dot_ (.) characters must be replaced with _underscore_ (_).
+For example, define the `swarm.bind.address=127.0.0.1` property as `SWARM_BIND_ADDRESS=127.0.0.1`.
+* All _dash/hyphen_ (-) characters must be replaced with the `_DASH_` string.
+For example, define the `swarm.data-sources.foo.url=<url>` property as `SWARM_DATA_DASH_SOURCES_FOO_URL=<url>`.
+* If the property name contains underscores, all _underscore_ (\_) characters must be replaced with the `_UNDERSCORE_` string.
+For example, define the `swarm.data_sources.foo.url=<url>` property as `SWARM_DATA_UNDERSCORE_SOURCES_FOO_URL=<url>`.
 
-Example Data Source Configuration, if your System properties are defined as
-====
- -Dswarm.datasources.data-sources.devwf.connection-url=jdbc:postgresql://localhost:5432/sampledb
- -Dswarm.datasources.data-sources.devwf.driver-name=postgresql
- -Dswarm.datasources.data-sources.devwf.jndiname=java:/jboss/datasources/devwf
- -Dswarm.datasources.data-sources.devwf.user-name=postgres
- -Dswarm.datasources.data-sources.devwf.password=admin
+.An example data source configuration
```

Comment:
I took the liberty to reformat the extensive configuration into tables that show the respective system property and environment variable side by side for clarity.